Remotion/Data/DomainObjects/Mapping/IMappingConfiguration.cs

```diff
@@ -24,11 +24,15 @@ namespace Remotion.Data.DomainObjects.Mapping
   /// </summary>
   public interface IMappingConfiguration
   {
-    ClassDefinition[] GetTypeDefinitions ();
-    bool ContainsTypeDefinition (Type classType);
-    ClassDefinition GetTypeDefinition (Type classType);
-    ClassDefinition GetTypeDefinition (Type classType, Func<Type, Exception> missingTypeDefinitionExceptionFactory);
+    TypeDefinition[] GetTypeDefinitions ();
+    ClassDefinition[] GetClassDefinitions ();
+    bool ContainsTypeDefinition (Type type);
+    TypeDefinition GetTypeDefinition (Type type);
+    TypeDefinition GetTypeDefinition (Type type, Func<Type, Exception> missingTypeDefinitionExceptionFactory);
+    bool ContainsClassDefinition (Type classType);
     bool ContainsClassDefinition (string classID);
+    ClassDefinition GetClassDefinition (Type type);
```

Comment:
The Type-based overloads shoudl be changed to extension methods.

GetClassDefinitions() is a big question mark. Do we need this method? And if so, where?

Reply:
The usages of `GetClassDefinitions` are now gone and I removed the method.